contracts/stable-asset/StableAsset.sol

```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./IStableAsset.sol";
+
+contract StableAsset is IStableAsset {
+    address constant private PRECOMPILE = address(0x0000000000000000000000000000000000000406);
+
+    /**
+     * @dev Get stable asset pool tokens.
+     * @param poolId ID of the pool
+     * @return poolExists whether pool exists or not
+     * @return tokens stabel asset pool tokens
+     */
+    function getStableAssetPoolTokens(uint32 poolId)
+    public
+    view
+    override
+    returns (bool, address[] memory) {
+        (bool success, bytes memory returnData) = PRECOMPILE.staticcall(
+            abi.encodeWithSignature("getStableAssetPoolTokens(uint32)", poolId)
+        );
+        assembly {
+            if eq(success, 0) {
+                revert(add(returnData, 0x20), returndatasize())
+            }
+        }
+        if (returnData.length == 0) {
+            // pool not found
+            address[] memory empty;
+            return (false, empty);
+        } else {
+            return (true, abi.decode(returnData, (address[])));
+        }
```

Comment:
Even if the pool exists, the caller has to check the response is not 0x0 address, so it feels like there is no need to return bool

Reply:
checking address is something else, bool indicates pool exists even if there's no tokens, empty array of addresses